### DISCUSSION TITLE
Which enhancement should be next?

### DISCUSSION THREAD

**Discussion by ManOfMeans** [Vail-CW/vail_repeater_depricated#59] [Polls]
Hello All!

To make the best use of Neale's time I thought it might be good to have a Poll to see which enhancement should be worked on next.  Please choose from the list provided.

**Comment by nealey** [comment]
Very excited to see how this poll turns out! This would in fact be helpful!

**Comment by nealey** [comment]
## Candidate in testing

Please give https://vail.woozle.org/testing/ a spin and let me know what you think.

## Vail is a telegraph, not FM radio

The sender can't tell the receiver what tone to play: it can only say "play a tone for this long". You can tell your tone generator the transmit and receive tone pitches, or you can use telegraph sounds for a more realistic experience.

If you are talking to four other people on the same repeater, all of the other people will have the same pitch.


## Why notes and not frequencies

The Vail adapter is a MIDI device, and MIDI wants notes, not frequencies. The sliders dial in a note, not a frequency. This is so you can have the same pitches on the adapter.

Interesting side note: you can also plug a MIDI synthesizer in to your computer and Vail will play on it! So if you want to do Morse Code with, like, a Yamaha DX7 "BRASS2" patch, you can.

**Reply by ManOfMeans** [reply]
Hi Neale,

Thank you for the test environment!  I am always impressed that you can whip up these customizations so quickly.
I have played with _testing_ a bit and have had mixed results.  I've developed some opinions through the process as well.  Here are my thoughts and findings.

### **Volume Slider**
This worked perfectly well!  I was able to adjust volume on both my phone and with my laptop's browser.  Slick!

### **TX**
TX tone C3 (131Hz) and below, I had no audio of this on my laptop.  I suppose my laptop's speaker doesn't have the fidelity that my phone does, because my phone was able to produce audible output at this range.  At about F#0 (23Hz) down to C-1 (8Hz) my phone only make audible "popping" when keying.  I suppose this might be useful for emulating how a telegraph sounds when you're operating one, but for practicality's sake and in my own opinion, about tone A3 (220Hz) is about the extreme low end of a practical CW sidetone.  I can't think of any reason to have one lower.

### **RX**
I had great difficulty with RX.  When I first loaded the page on my cell and laptop to test between the two, I had no problem hearing both devices.  However, after the first or second time I made an RX adjustment, I lost the ability to hear the incoming signal entirely.  This happened with both devices, and was only rectified after I closed the browser on both devices.  I was able to reproduce this process twice, Google Chrome on the laptop and also Chrome on the iPhone.  Not sure what's happening here.

I tested again, this time using Microsoft Edge on the laptop and Apple's Safari on the iPhone.  Without making any changes to TX or RX, I could see the lamp illuminate when the sending station transmits, but there was absolutely no RX audio on either device.  When I browsed to the "production" Vail site (non-testing), I was able to hear the RX without issue.

Synopsis:  When sending from testing, I can hear the sender while I'm in the production site.  When sending from the production site, those in testing cannot hear.  When sending in testing to another user in testing, nothing is heard either except for the first time you join the site.
Volume slider works well!

I hope this helps!  Please let me know if you'd need anything...

**Reply by nealey** [reply]
## Volume

Awesome. I sort of implemented that by mistake, I thought that's what the poll said was most requested. Oh well :smile: 

## Tones below F#0

I used to think 40WPM was fast enough, but some crazy guy is now going at 50, and I have learned not to make any assumptions. I'm going to leave the full range of MIDI notes, but feel free not to use all of them :wink: 

## RX dropout

Man, I hate bugs like this :angry: Are you able to reproduce this with the fortunes repeater, or does it have to be live chatter?

**Reply by nealey** [reply]
I think I found the problem! Hold on a few minutes and I'll push a new version up.

**Reply by nealey** [reply]
Okay, I think I've fixed this in https://github.com/nealey/vail/commit/6ebb9e621dcef2b1cda2ead0e5fb302985275391.

Please give it a try in testing and let me know if you still see the dropped rx (or any other problems)!

**Reply by ManOfMeans** [reply]
Okay!  This works!  Tested in Chrome on mobile and on computer.  I can key and be heard, even after altering TX and RX notes, and also while keying across the production and testing platforms.  All checks out here.

I also asked for help to test this Live on the testing site and Gunner (KK7DEU) and Paul (N6EV) joined me there.  I am happy to report we had a successful QSO and were able to use the RX, TX and Volume sliders without issue.

Neale, you nailed this! 👍

**Reply by nealey** [reply]
Great! Pushed into production.
